packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3848,7 +3848,9 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
         if (paintBounds.bottom <= box.top) {
           break;
         }
-        if (paintBounds.contains(Offset(box.left, box.top)) || paintBounds.contains(Offset(box.right, box.bottom))) {
+        // Include any rect with overlap with paintBounds.
+        if ((paintBounds.left <= box.right || paintBounds.right >= box.left) &&
```

Comment:
should this be be `&&` instead of `||`?

Reply:
No, it's correct. If `paintBounds.left <= box.right`, then we have some horizontal overlap on the left side of paintBounds. And if `paintBounds.right >= box.left`, we have some horizontal overlap on the right side of paintBounds.

Reply:
`paintBounds.left <= box.right` doesn't seem to guarantee horizontal overlap since the `box` can still be too far to the right? 

The current expression is equivalent to `!(box.right < paintBounds.left && paintBounds.right < box.left)`. Since `paintBounds.left <= paintBounds.right` the expression is almost always true.

Reply:
I'm still a bit confused on this (maybe i'm missing some context). Could you explain the purpose of these 2 fields? (paintBounds and box)

Reply:
`paintBounds` is the area of the text field. `box` is the area of some letter. To decide which letters to send to the engine, this change will accept letters with any overlap with the text field. Since that means they will be visible to the user. Checking only that the top-left or bottom-right corner of the letter was inside the text field missed some cases (tall letters, letters near the edge).